### PR TITLE
wip

### PR DESCRIPTION
**Pull request by maliheh-deriv**
## Description 
This card aim to migrate the test cases regarding compare links in RF to the cypress code. 
compare links  - EU 
compare links -  DIEL 
compare links - Row 

## cards 
https://app.clickup.com/t/20696747/QATEST-97047
https://app.clickup.com/t/20696747/DPROD-3500

### CODE REVIEW

File: cypress/e2e/full/comparelinksDIEL.cy.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  len = a.length
+  cy.log('the length is ',len)
+  .then(() => {
+    // Remove duplicates using a Set
+    CurrentLinks = Array.from(new Set(IinitialList));
+    cy.log('Unique Links:', CurrentLinks);
+    cy.log('Unique Array:', CurrentLinks.join(', '));
+  });
+})});
+});
```
Comment (bilalmajid-deriv) [inline cypress/e2e/full/comparelinksDIEL.cy.js:26]: Suggestion: It is better to place the before hooks between the body of the relevant describe()

File: cypress/e2e/full/comparelinksDIEL.cy.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+
+describe('QATEST-97047 - Compare URL in production and staging', () => {
+
+  it('visits all links for DIEL countries in staging and compare links with pre-defined list', () => {
```
Comment (bilalmajid-deriv) [inline cypress/e2e/full/comparelinksDIEL.cy.js:31]: CHANGE: The test case name should be starting with should like
it('should do something........',()={

as available in coding standards section in https://wikijs.deriv.cloud/en/Quality-Assurance/e2e-test-automation

**Review by bilalmajid-deriv — COMMENTED** [review]
Looks good to me with one change and some suggestions.

Suggestion:
As the tests cases are the same in all the files the only difference between them is the region I think we can move all three it() in one single describe in one test file and name the tests compareLinks.cy.js and then we can move the duplicate lines of code in one single parameterized function in which we can give all the different values such as region or the list of links for each test as a parameter. or we can create just on it() and run multiple test using that one it(), this will pragmatically create tests. The later would be more tricky

**Review by bilalmajid-deriv — COMMENTED** [review]
LGTM

**Review by markodonnell-deriv — APPROVED** [review]
Looks good. But should the staging links be hardcoded?
I'll approve and merge for now so we can see if it runs ok in Cypress Cloud.

**Pull request merged**